thorlcr/thorutil/thormisc.cpp

```diff
@@ -1670,3 +1670,31 @@ StringBuffer &getExpertOptString(const char *opt, StringBuffer &out)
     globals->getProp(xpath, out);
     return out;
 }
+
+// NB: in practice this is for containerized only
+void applyResourcedCPUs(const char *resourceSection)
+{
+    unsigned maxActivityCores = getExpertOptInt64(THOROPT_MAX_ACTIVITY_CORES);
+    if (maxActivityCores)
+        return;
+    VStringBuffer xpath("%s/@cpu", resourceSection);
+    if (!globals->hasProp(xpath))
+        return;
+    const char *cpusText = globals->queryProp(xpath);
+    double cpus = friendlyCPUToDecimal(cpusText);
+    if (0.0 == cpus)
+        throw makeStringExceptionV(0, "Invalid number of %s.cpu: %s", resourceSection, cpusText);
+    unsigned __int64 cpusI = (unsigned __int64)cpus;
+    if (cpus != (double)cpusI)
+    {
+        OWARNLOG("Fractional number of CPUs '%s' specified in %s.cpu. May cause poor performance", cpusText, resourceSection);
+        // round up
```

Comment:
Is rounding up a good idea?  It means it will probably throttle at some point.

Reply:
fractional is a bad idea full stop, we should probably reject it at the helm level.
Could change to round down..